scripts/code.angularjs.org-firebase/functions/index.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+const functions = require('firebase-functions');
+const gcs = require('@google-cloud/storage')();
+const path = require('path');
+
+const gcsBucket = 'code-angularjs-org-338b8.appspot.com';
+const LOCAL_TMP_FOLDER = '/tmp/';
+
+function sendStoredFile(request, response) {
+  let filePathSegments = request.path.split('/').filter((segment) => {
+    return segment !== '';
+  });
+
+  const version = filePathSegments[0];
+  const isDocsPath = filePathSegments[1] === 'docs';
+  const lastSegment = filePathSegments[filePathSegments.length - 1];
+  let downloadPath;
+  let fileName;
+
+  if (isDocsPath && filePathSegments.length === 2) {
+    fileName = 'index.html';
+    filePathSegments = [version, 'docs', fileName];
+  } else {
+    fileName = lastSegment;
+  }
+
+  downloadPath = path.join.apply(null, filePathSegments);
+
+  const bucket = gcs.bucket(gcsBucket);
+
+  downloadAndSend().catch(error => {
+    if (isDocsPath && error.code === 404) {
+      fileName = 'index.html';
+      filePathSegments = [version, 'docs', fileName];
+      downloadPath = path.join.apply(null, filePathSegments);
+      return downloadAndSend();
+    }
+
+    return Promise.reject(error);
+  }).catch(error => {
+    let message = 'General error';
+    if (error.code === 404) {
+      if (fileName.split('.').length === 1) {
+        message = 'Directory listing is not supported';
```

Comment:
Could we host directly from Google storage (something like [this][1])?

[1]: https://cloud.google.com/storage/docs/hosting-static-website